### ISSUE TITLE
Show a custom overlay on DataGrid when there has been an error.

### ISSUE THREAD

**Issue by john1625b** [mui/mui-x#10053]
### Order ID

72840

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

I know that custom overlays exist for [loading and now rows](https://mui.com/x/react-data-grid/components/#loading-overlay) 
 
 
 However, I need a custom message on the table to tell the users that the network request to get the table data has failed. The "No rows" message would be misleading since 0 rows is different from the data failing to be fetched. 
 
What's the best way of accomplishing this? 

Thanks

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
System:
    OS: macOS 13.5
  Binaries:
    Node: 18.16.1 - ~/.nvm/versions/node/v18.16.1/bin/node
    Yarn: 1.22.18 - /opt/homebrew/bin/yarn
    npm: 9.8.1 - ~/.nvm/versions/node/v18.16.1/bin/npm
  Browsers:
    Chrome: 115.0.5790.114
    Edge: Not Found
    Safari: 16.6
  npmPackages:
    @emotion/react: ^11.11.1 => 11.11.1 
    @emotion/styled: ^11.11.0 => 11.11.0 
    @mui/base:  5.0.0-beta.8 
    @mui/core-downloads-tracker:  5.14.1 
    @mui/icons-material: ^5.14.1 => 5.14.1 
    @mui/material: ^5.14.1 => 5.14.1 
    @mui/private-theming:  5.13.7 
    @mui/styled-engine:  5.13.2 
    @mui/system:  5.14.1 
    @mui/types:  7.2.4 
    @mui/utils:  5.14.5 
    @mui/x-data-grid:  6.11.1 
    @mui/x-data-grid-generator: ^6.11.1 => 6.11.1 
    @mui/x-data-grid-premium: ^6.10.2 => 6.11.1 
    @mui/x-data-grid-pro:  6.11.1 
    @mui/x-license-pro:  6.10.2 
    @types/react: ^18.2.15 => 18.2.15 
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.2.0 
    typescript: ^5.0.2 => 5.1.6 
```
</details>

**Comment by romgrk** [comment]
Have you considered adding an overlay outside of the grid? It would be the easiest way to solve your issue.

You could also fork the [custom overlay example](https://mui.com/x/react-data-grid/components/#no-rows-overlay) to show an error message conditionally, if applicable.

**Comment by john1625b** [comment]
> Have you considered adding an overlay outside of the grid? 

As in doing something like this`!error ? <DataGrid.../> : <div> Error! </div>`

that could work but it would be more convenient to just pass a prop. It would also make it easier to enforce an opininated way to handle errors with DataGrid to the rest of my coworkers.

**Comment by romgrk** [comment]
I don't think we would add a new prop for that purpose, you can already create a customized message overlay.

I think this is what you want: https://codesandbox.io/s/gifted-rui-c78zmf?file=/Demo.js

**Comment by john1625b** [comment]
@romgrk That's a good solution. Thanks.
